packages/zquery/src/index.test.ts

```diff
@@ -1,57 +1,222 @@
-  type MockState = null;
-  let mockUseStore: UseStore<MockState>;
-  let mockState: MockState;
+  let zQuery: ZQuery<'puppyPhotos', PuppyPhoto[], []>;
+  let useStore: UseBoundStore<StoreApi<State>>;
+  const fetch = vi.fn().mockResolvedValue(MOCK_PUPPY_PHOTOS);
 
   beforeEach(() => {
-    mockState = null;
-    mockUseStore = Object.assign(<T>(selector: (state: MockState) => T) => selector(mockState), {
-      getState: () => mockState,
-    });
-  });
+    fetch.mockClear();
 
-  describe('the return value', () => {
-    const result = createZQuery({
+    zQuery = createZQuery({
       name: 'puppyPhotos',
-      fetch: () => Promise.resolve(null),
-      getUseStore: () => mockUseStore,
-      set: () => {
-        /* no-op */
+      fetch,
+      getUseStore: () => useStore,
+      get: state => state.puppyPhotos,
+      set: setter => {
+        const newState = setter(useStore.getState().puppyPhotos);
+        useStore.setState(state => ({
+          ...state,
+          puppyPhotos: newState,
+        }));
       },
-      get: () => ({
-        _zQueryInternal: {
-          fetch: vi.fn(),
-          referenceCount: 0,
-        },
-        loading: false,
-        revalidate: vi.fn(),
-        data: undefined,
-        error: undefined,
-      }),
     });
 
+    const { puppyPhotos } = zQuery;
+
+    useStore = create<State>()(() => ({
+      puppyPhotos,
+    }));
+  });
+
+  describe('the return value', () => {
     it('includes hooks and a Zustand slice that use the passed-in name', () => {
-      expect(result).toHaveProperty('puppyPhotos');
-      expect(result.usePuppyPhotos).toBeTypeOf('function');
-      expect(result.useRevalidatePuppyPhotos).toBeTypeOf('function');
+      expect(zQuery).toHaveProperty('puppyPhotos');
+      expect(zQuery.usePuppyPhotos).toBeTypeOf('function');
+      expect(zQuery.useRevalidatePuppyPhotos).toBeTypeOf('function');
     });
 
-    it('includes the correct default state', () => {
-      expect(result.puppyPhotos).toEqual({
-        data: undefined,
-        error: undefined,
-        loading: false,
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-        revalidate: expect.any(Function),
-        _zQueryInternal: {
+    describe('the ZQuery slice', () => {
+      it('has the correct default state', () => {
+        expect(zQuery.puppyPhotos).toEqual({
+          data: undefined,
+          error: undefined,
+          loading: false,
           // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-          fetch: expect.any(Function),
-          referenceCount: 0,
-        },
+          revalidate: expect.any(Function),
+          _zQueryInternal: {
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+            fetch: expect.any(Function),
+            referenceCount: 0,
+          },
+        });
+      });
+
+      describe('the reducers returned by the `stream()` function', () => {
```

Comment:
The tests in this suite are all new, and they test the various `on*()` reducers that you can pass to `createZQuery()`.